machine/quanta/quanta_common_p2020/u-boot/platform-quanta-common_p2020.patch

```diff
@@ -0,0 +1,2235 @@
++QUANTA_COMMON_P2020          powerpc     mpc85xx     quanta_common_p2020          quanta
++QUANTA_COMMON_P2020-R0       powerpc     mpc85xx     quanta_common_p2020          quanta         -           QUANTA_COMMON_P2020:ONIE_PLATFORM_REV=0
+ P2020RDB                     powerpc     mpc85xx     p1_p2_rdb           freescale      -           P1_P2_RDB:P2020RDB
+ P2020RDB_36BIT               powerpc     mpc85xx     p1_p2_rdb           freescale      -           P1_P2_RDB:P2020RDB,36BIT
+ P2020RDB_36BIT_SDCARD        powerpc     mpc85xx     p1_p2_rdb           freescale      -           P1_P2_RDB:P2020RDB,36BIT,SDCARD
+diff --git a/common/cmd_nvedit.c b/common/cmd_nvedit.c
+index 7633f0c..ab20665 100644
+--- a/common/cmd_nvedit.c
++++ b/common/cmd_nvedit.c
+@@ -295,6 +295,36 @@ int setenv_ulong(const char *varname, ulong value)
+ }
+ 
+ /**
++ * Set an environment variable to an value in hex
++ *
++ * @param varname	Environment variable to set
++ * @param value		Value to set it to
++ * @return 0 if ok, 1 on error
++ */
++int setenv_hex(const char *varname, ulong value)
++{
++	char str[17];
++
++	sprintf(str, "%lx", value);
++	return setenv(varname, str);
++}
++
++ulong getenv_hex(const char *varname, ulong default_val)
++{
++	const char *s;
++	ulong value;
++	char *endp;
++
++	s = getenv(varname);
++	if (s)
++		value = simple_strtoul(s, &endp, 16);
++	if (!s || endp == s)
++		return default_val;
++
++	return value;
++}
++
++/**
+  * Set an environment variable to an address in hex
+  *
+  * @param varname	Environmet variable to set
+diff --git a/common/cmd_sys_eeprom.c b/common/cmd_sys_eeprom.c
+index 3d5dace..2309974 100644
+--- a/common/cmd_sys_eeprom.c
++++ b/common/cmd_sys_eeprom.c
+@@ -9,6 +9,7 @@
+ #include <command.h>
+ #include <i2c.h>
+ #include <linux/ctype.h>
++#include <mb_detect.h>
+ 
+ #include "sys_eeprom.h"
+ 
+@@ -23,7 +24,6 @@ static bool tlvinfo_find_tlv(u8 * eeprom, u8 tcode, int * eeprom_index);
+ static bool tlvinfo_delete_tlv(u8 * eeprom, u8 code);
+ static bool tlvinfo_add_tlv(u8 * eeprom, int tcode, char * strval);
+ static int set_mac(char *buf, const char *string);
+-static int set_date(char *buf, const char *string);
+ static int set_bytes(char *buf, const char *string, int * converted_accum);
+ 
+ /* Set to 1 if we've read EEPROM into memory */
+@@ -175,6 +175,7 @@ static void show_eeprom(u8 *eeprom)
+ 	printf("Checksum is %s.\n", is_checksum_valid(eeprom) ? "valid" : "invalid");
+ 
+ #ifdef DEBUG
++	int i;
+ 	printf("EEPROM dump: (0x%x bytes)", TLV_INFO_MAX_LEN);
+ 	for (i = 0; i < TLV_INFO_MAX_LEN; i++) {
+ 		if ((i % 16) == 0)
+@@ -214,7 +215,7 @@ static struct tlv_code_desc tlv_code_list[] = {
+ 	{ TLV_CODE_VENDOR_NAME	 , "Vendor Name"},
+ 	{ TLV_CODE_DIAG_VERSION	 , "Diag Version"},
+ 	{ TLV_CODE_SERVICE_TAG   , "Service Tag"},
+-	{ TLV_CODE_VENDOR_EXT	 , "Vendor Extension"},
++	{ TLV_CODE_VENDOR_EXT	 , "Model Name" /* "Vendor Extension" */},
```

Comment:
This cannot be done.

`TLV_CODE_VENDOR_EXT` cannot simply be renamed and used for a different purpose.  The EEPROM spec is a standard and 3rd parties rely on it being consistent.

If the need arises to define a new TLV then that is separate topic we should discuss on the mailing.  The first thing to do is make absolutely sure none of the existing TLVs will work for the project:
https://github.com/opencomputeproject/onie/wiki/Design-Spec-HW-Requirements#type-code-values